ogre2/src/Ogre2WideAngleCamera.cc

```diff
@@ -0,0 +1,678 @@
+#include <Compositor/Pass/PassScene/OgreCompositorPassSceneDef.h>
+#include <OgreDepthBuffer.h>
+#include <OgreImage2.h>
+#include <OgrePass.h>
+#include <OgreRoot.h>
+#include <OgreSceneManager.h>
+#include <OgreTechnique.h>
+#include <OgreTextureBox.h>
+#include <OgreTextureGpuManager.h>
+#ifdef _MSC_VER
+#  pragma warning(pop)
+#endif
+
+// clang-format off
+namespace gz
+{
+namespace rendering
+{
+inline namespace GZ_RENDERING_VERSION_NAMESPACE {
+/// \brief Helper class for setting up Camera and Materials when rendering
+/// via Ogre2WideAngleCamera.
+class GZ_RENDERING_OGRE2_HIDDEN Ogre2WideAngleCameraWorkspaceListenerPrivate :
+    public Ogre::CompositorWorkspaceListener
+{
+  public: gz::rendering::Ogre2WideAngleCamera &owner;
+
+  /// \brief constructor
+  /// \param[in] _scene the scene manager responsible for rendering
+  public: explicit Ogre2WideAngleCameraWorkspaceListenerPrivate(
+        gz::rendering::Ogre2WideAngleCamera &_owner) :
+    owner(_owner)
+  {
+  }
+
+  /// \brief Called when each pass is about to be executed.
+  /// \param[in] _pass Ogre pass which is about to execute
+  public: virtual void passPreExecute(Ogre::CompositorPass *_pass) override;
+};
+}
+}
+}
+// clang-format on
+
+static const uint32_t kWideAngleNumCubemapFaces = 6u;
+
+/// \brief Private data for the WideAngleCamera class
+class gz::rendering::Ogre2WideAngleCamera::Implementation
+{
+  // clang-format off
+  /// \brief Environment texture size
+  public: uint32_t envTextureSize = 512u;
+
+  /// \brief A single cube map texture
+  public: Ogre::TextureGpu *envCubeMapTexture = nullptr;
+
+  /// \brief Output texture
+  public: Ogre::TextureGpu *ogreRenderTexture = nullptr;
+
+  /// \brief Compositor workspace. Does all the work
+  public: Ogre::CompositorWorkspace *ogreCompositorWorkspace = nullptr;
+
+  /// \brief Main pass definition (used for visibility mask manipuluation).
+  public: Ogre::CompositorPassSceneDef
+  *cubePassSceneDef[kWideAngleNumCubemapFaces]{};
+
+  /// \brief Pointer to material, used for second rendering pass
+  public: Ogre::MaterialPtr compMat;
+
+  /// \brief Camera lens description
+  public: CameraLens lens;
+
+  /// \brief Pointer to the ogre camera
+  /// OgreNext rotates to face each of the 6 sides
+  /// We use listeners to set FOV to the proper setting depending on the pass
+  public: Ogre::Camera *ogreCamera = nullptr;
+
+  /// \brief Dummy texture
+  public: Ogre2RenderTexturePtr wideAngleTexture;
+
+  /// \brief Event used to signal camera data
+  public: gz::common::EventT<void(const unsigned char *,
+              unsigned int, unsigned int, unsigned int,
+              const std::string &)> newImageFrame;
+
+  /// \brief See Ogre2WideAngleCameraWorkspaceListenerPrivate
+  public: Ogre2WideAngleCameraWorkspaceListenerPrivate workspaceListener;
+
+  explicit Implementation(gz::rendering::Ogre2WideAngleCamera &_owner) :
+    workspaceListener(_owner)
+  {
+  }
+  // clang-format on
```

Comment:
Out of curiosity, do you have a clang-format config that approaches the gazebo style?

Reply:
Email sent